ci/clean.py

```diff
@@ -123,6 +123,8 @@ def _enumerate_objects_from_release_manifest_sets(
     for release_manifest_set in enumerate_release_sets(prefix=prefix):
         for release_manifest in release_manifest_set.manifests:
             yield from release_manifest.paths
+        if release_manifest_set.logs:
```

Comment:
if `logs` be initialised w/ an empty iterable (e.g. the empty tuple), this special-case-handling would not be required. Also: how about moving the logic of determining all relevant / referenced blobs to the `Manifest` dataclass?

Reply:
w.r.t moving this code to dataclass: I guess this needs more discussion, will put in on the agenda